packages/flame/lib/src/components/position_component.dart

```diff
@@ -80,7 +86,9 @@ abstract class PositionComponent extends BaseComponent {
 
   /// Get the position of the center of the component's bounding rectangle without rotation
   Vector2 get center {
-    return anchor == Anchor.center ? position : topLeftPosition + (size / 2);
+    return anchor == Anchor.center
+        ? position
+        : anchor.toOtherAnchorPosition(position, Anchor.center, size);
```

Comment:
if anchor = center it should work as well right? no need for the ternary?

Reply:
It would, but since this is called quite often I think it is good to have this optimization.